src/ars/migrations/definitions.py

```diff
@@ -86,8 +95,75 @@ async def remove_access_request_fields(doc: Document) -> Document:
                 del doc[field]
             return doc
 
-        async with self.auto_finalize(coll_names="accessRequests", copy_indexes=False):
+        async with self.auto_finalize(coll_names=ACCESS_REQUESTS, copy_indexes=False):
             await self.migrate_docs_in_collection(
-                coll_name="accessRequests",
+                coll_name=ACCESS_REQUESTS,
                 change_function=remove_access_request_fields,
             )
+
+
+class V3Migration(MigrationDefinition, Reversible):
+    """Apply field type updates for hexkit v6
+
+    Affected data:
+    - Collection: `accessRequests`:
+      - `_id`, `user_id`, `iva_id`, `changed_by`, `__metadata__.correlation_id`: str -> UUID
+        - this only affects non-null values for `changed_by` & `iva_id`
+      - `access_starts`, `access_ends`, `request_created`, `status_changed`: str -> datetime
+        - only applies to non-null values for `status_changed`
+    """
+
+    version = 3
+
+    async def apply(self):
+        """Apply the migration"""
+        _convert_required_fields = convert_uuids_and_datetimes_v6(
+            uuid_fields=["_id", "user_id"],
+            date_fields=["access_starts", "access_ends", "request_created"],
+        )
+        _convert_optional_dates = convert_uuids_and_datetimes_v6(
+            date_fields=["status_changed"]
+        )
+
+        async def _convert_doc(doc: Document) -> Document:
+            """Convert field types from string to UUID/datetime"""
+            doc = await convert_outbox_correlation_id_v6(doc)
+            doc = await _convert_required_fields(doc)
+            for field in ["iva_id", "changed_by"]:
+                if optional_uuid := doc[field]:
+                    doc[field] = UUID(optional_uuid)
+            # changed_by and status_changed are probably always populated together, but
+            #  no reason to embed that assumption here
+            if doc["status_changed"]:
+                doc = await _convert_optional_dates(doc)
```

Comment:
I would do it in the same way as for the optional UUIDs, that's easier to understand.

Would have been better with a special `convert_optional_uuids_and_datetimes_v6` in hexkit.